File: jekyoung/week6/11047.py
N, K = map(int, input().split())
coins = []
cnt = 0

# 입력받은 동전 가치를 리스트에 저장
for i in range(N):
    coins.append(int(input()))

# 가장 큰 가치의 동전부터 사용한다.
for j in range(N-1, -1, -1):
    # 남아있는 금액보다 동전의 가치가 작거나 같을 때, 지불 가능
    if K >= coins[j]:
        cnt += K//coins[j]
        K = K%coins[j]

print(cnt)

File: youngen/week6/11047.js
const fs = require('fs');
const input = fs.readFileSync("youngen/week6/input.txt").toString().trim().split("\n");

const [a, ...b] = input
const n = +a.split(' ')[0]
const k = +a.split(' ')[1]
const moneyVariables = b.map((item) => +item).reverse()
const moneyArr = [] //사용가능한 동전

for(let i = 0; i < n ; i++){
  if( k/ moneyVariables[i] >= 1){
    moneyArr.push(moneyVariables[i])
  }
}
let rest = k
let sum = 0
moneyArr.forEach((item)=>{
const gae = Math.floor(rest / item)
 rest = rest -  item * gae
 sum += gae
})
console.log(sum)
